### PR TITLE
Update json_value.cpp, fixing up spelling mistakes (boundry to boundary)

### PR DESCRIPTION
**Pull request by Guguant**
Here is a common spelling mistake that I've found while fixing up spelling mistakes.

### CODE REVIEW

File: src/lib_json/json_value.cpp [hunk 1]
```diff
@@ -1401,7 +1401,7 @@ void Value::Comments::set(CommentPlacement slot, String comment) {
   if (!ptr_) {
     ptr_ = std::unique_ptr<Array>(new Array());
   }
-  // check comments array boundry.
+  // check comments array boundary.
```
Comment (BillyDonahue) [inline src/lib_json/json_value.cpp:1404]: Would also need capitalization. Let's just remove the comment.
It's fairly obvious why the if precedes the array element assignment.

Comment (Guguant) [reply src/lib_json/json_value.cpp]: ok

File: src/lib_json/json_value.cpp [hunk 2]
```diff
@@ -1401,7 +1401,7 @@ void Value::Comments::set(CommentPlacement slot, String comment) {
   if (!ptr_) {
     ptr_ = std::unique_ptr<Array>(new Array());
   }
-  // check comments array boundry.
+
```
Comment (BillyDonahue) [inline src/lib_json/json_value.cpp:1404]: I don't really want the blank line either.
But while we're manipulating this function let's make the change more meaningful.

```c++
void Value::Comments::set(CommentPlacement slot, String comment) {
  if (slot >= CommentPlacement::numberOfCommentPlacement)
    return;
  if (!ptr_)
    ptr_ = std::unique_ptr<Array>(new Array());
  (*ptr_)[slot] = std::move(comment);
}
```

In other words, if we're not actually going to write to `ptr_`, we don't need to make one. We don't need to allocate it until we actually use it, and errant calls to `set` will then have no effect, which is a cleaner recovery I think.

Comment (Guguant) [reply src/lib_json/json_value.cpp]: I agree with your opinion.

File: src/lib_json/json_value.cpp [hunk 3]
```diff
@@ -1398,11 +1398,11 @@ String Value::Comments::get(CommentPlacement slot) const {
 }
 
 void Value::Comments::set(CommentPlacement slot, String comment) {
+  if (slot >= CommentPlacement::numberOfCommentPlacement) {
```
Comment (BillyDonahue) [inline src/lib_json/json_value.cpp:1401]: This is not the code I suggested. Can you do it again?

Comment (Guguant) [reply src/lib_json/json_value.cpp:1401]: Yes. If return directly while `slot >= CommentPlacement::numberOfCommentPlacement`, the **CI** told that Some checks were not successful.

Comment (BillyDonahue) [reply src/lib_json/json_value.cpp]: I'd be interested in the CI failure.

**Comment by BillyDonahue** [comment]
I merged my version of this. Thanks for bringing it up.

**Pull request closed without merging**